mmdet/models/backbones/resnet.py

```diff
@@ -312,13 +355,111 @@ def train(self, mode=True):
         if mode and self.frozen_stages >= 0:
             for param in self.conv1.parameters():
                 param.requires_grad = False
-            for param in self.bn1.parameters():
+
+            stem_norm = getattr(self, self.stem_norm_name)
+            stem_norm.eval()
+            for param in stem_norm.parameters():
                 param.requires_grad = False
-            self.bn1.eval()
-            self.bn1.weight.requires_grad = False
-            self.bn1.bias.requires_grad = False
+
             for i in range(1, self.frozen_stages + 1):
                 mod = getattr(self, 'layer{}'.format(i))
                 mod.eval()
                 for param in mod.parameters():
                     param.requires_grad = False
+
+
+class ResNetClassifier(ResNet):
```

Comment:
This class may not be necessary now. We can later add an option to include fc layers in ResNet backbone.